src/WebJobs.Script/Config/INameResolverExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Host;
+using System;
+using System.Reflection;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    // Attribute to mark a string property has embedded %key% and supports INameResolver resolution. 
+    [AttributeUsage(AttributeTargets.Property)]
+    public class AllowNameResolutionAttribute : Attribute
```

Comment:
Move this into its own class. Actually I've enabled fxcop/stylecop on all projects, so when you sync/rebase, I'd expect this to get flagged